subprojects/platform-jvm/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaToolchainQueryServiceTest.groovy

```diff
@@ -304,6 +308,66 @@ class JavaToolchainQueryServiceTest extends Specification {
         toolchain.getInstallationPath().toString() == systemSpecificAbsolutePath("/path/1.8.2")
     }
 
+    def "nag user about invalid toolchain spec when #description"() {
```

Comment:
We normally do integration tests for deprecation warnings. Check out `GradleExecuter.expectDeprecationWarning`.

Reply:
I've seen some test that do it in unit tests. Made the wrong conclusion.
Fixed it now.